pkg/sysload/cpu.go

```diff
@@ -45,6 +51,63 @@ func Init(conf *config.Config, close chan struct{}, isAvailable func() float64)
 	prometheus.MustRegister(promNodeAvailable)
 
 	go monitorCPULoad(close)
+	return nil
+}
+
+func checkCPUConfig(costConfig config.CPUCostConfig) error {
+	if costConfig.RoomCompositeCpuCost < 2.5 {
+		logger.Warnw("room composite requirement too low", nil,
+			"config value", costConfig.RoomCompositeCpuCost,
+			"recommended minimum value", 2.5,
+		)
+	}
+	if costConfig.TrackCompositeCpuCost < 1 {
+		logger.Warnw("track composite requirement too low", nil,
+			"config value", costConfig.TrackCompositeCpuCost,
+			"recommended minimum value", 1,
+		)
+	}
+	if costConfig.TrackCpuCost < 1 {
+		logger.Warnw("track requirement too low", nil,
+			"config value", costConfig.RoomCompositeCpuCost,
+			"recommended minimum value", 1,
+		)
+	}
+
+	requirements := []float64{
+		costConfig.TrackCpuCost,
+		costConfig.TrackCompositeCpuCost,
+		costConfig.RoomCompositeCpuCost,
+	}
+	sort.Float64s(requirements)
+
+	recommendedMinimum := requirements[2]
+	if recommendedMinimum < 3 {
+		recommendedMinimum = 3
+	}
+
+	if numCPUs < requirements[0] {
+		recommended := requirements[2]
+		if recommended < 3 {
+			recommended = 3
+		}
+		logger.Errorw("not enough cpu", nil,
+			"minimum cpu", requirements[0],
+			"recommended", recommended,
+			"available", numCPUs,
+		)
+		return errors.New("not enough cpu")
+	}
+
+	if numCPUs < requirements[2] {
```

Comment:
nit: this seems like an assumption that RoomComposite will always cost more.. it's a fair assumption, but I'm always wary about stuff like this.

it would also be better if we printed explicitly which requests this node will not be able to support. IMO it's more clear for someone trying to start a RoomComposite to see that it is explicitly not supported.

Reply:
they're sorted, it's not making that assumption

Reply:
sorry, I missed that line